File: 2015/ruby/day14/part1.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        reindeer = []
        max = 0

        file.each_line do |line|
            sentence = line.split(' ')
            reindeer.push({
                'speed' => sentence[3].to_i,
                'flyTime' => sentence[6].to_i,
                'restTime' => sentence[13].to_i
            })
        end

        reindeer.each do |stats|
            period = stats['flyTime'] + stats['restTime']
            flightTime = (2503 / period) * stats['flyTime']
            flightTime += [2503 % period, stats['flyTime']].min
            max = [flightTime * stats['speed'], max].max
        end

        puts max
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day20/part1.py
#!/usr/bin/python

with open('input.txt') as file_handler:
    ranges = [map(int, line.rstrip().split('-')) for line in file_handler]

ip = 0
upper = 0

while upper is not None:
    ip = upper + 1
    upper = next((rng[1] for rng in ranges if rng[0] <= ip <= rng[1]), None)

print ip
